### DISCUSSION TITLE
flow to run private examples

### DISCUSSION THREAD

**Discussion by nkint** [thi-ng/umbrella#474] [General]
Hi Karsten,
this is a nooby one 😅 sorry for asking some stupid thing but it's a while I'm not chilling around and I've seen a lot of new changes.
I'm looking for a workflow to create private examples and try new juicy packages...

Note that, of course, it is something I can easily do in a couple of minutes by hand but I'm sure you have an automated way to do it with tools and scripts so I have to miss something.

Let's try to do a fresh new example called `subdivision-from-waveform`.

my input:
```sh
 umbrella git:(develop)$   sh ./scripts/make-example subdivision-from-waveform
```

output:
```sh
generating module: examples/subdivision-from-waveform
creating /src folder...
creating /css folder...
writing package.json...
writing tsconfig.json...
writing vite.config.js...
writing index.html...
writing README.md...
refreshing monorepo index...
➤ YN0000: · Yarn 4.2.2
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: · Done in 0s 499ms
DONE
```

my input:
```sh
umbrella git:(develop)$   mv examples/subdivision-from-waveform examples-private
umbrella git:(develop)$   cd examples-private/subdivision-from-waveform
subdivision-from-waveform git:(develop)$   yarn start
```

output:
```sh
Usage Error: The nearest package directory (/Users/alberto/Code/umbrella/examples-private/subdivision-from-waveform) doesn't seem to be part of the project declared in /Users/alberto/Code/umbrella.

- If /Users/alberto/Code/umbrella isn't intended to be a project, remove any yarn.lock and/or package.json file there.
- If /Users/alberto/Code/umbrella is intended to be a project, it might be that you forgot to list examples-private/subdivision-from-waveform in its workspace configuration.
- Finally, if /Users/alberto/Code/umbrella is fine and you intend examples-private/subdivision-from-waveform to be treated as a completely separate project (not even a workspace), create an empty yarn.lock file in it.

$ yarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] [--require #0] <scriptName> ...
```

**Comment by postspectacular** [comment]
Hi @nkint — I can't even remember why I checked in that `examples-private` folder into Git. I'm only using it as a scratch pad for moving in WIP examples which aren't ready for publishing yet...

I _think_ you can use it the way you want though by editing the main/root-level `package.json` to include that folder into the `workspaces` list:

https://github.com/thi-ng/umbrella/blob/6ab7774f83bf4300738b00fc2ba947c18c7b2b7a/package.json#L4-L8

After making the change, you'll probably have to run `yarn` (without args) one more time... Hope that helps! Also, if it doesn't just leave the example in the normal `examples` directory, no harm done... :)

**Reply by nkint** [reply]
ok, thanks
